docs/site/Deploying-to-IBM-Cloud.md

```diff
@@ -0,0 +1,211 @@
+
+```sh
+$ npm i cfenv -s
+```
+
+Update the `index.ts` file to the following to enable service binding.
+
+```ts
+import {TodoListApplication} from './application';
+import {ApplicationConfig} from '@loopback/core';
+const datasourceDb = require('./datasources/db.datasource.json');
+const cfenv = require('cfenv');
+const appEnv = cfenv.getAppEnv();
+if (!appEnv.isLocal) {
+  // 'myCloudant' is the name of the provisioned Cloudant service
+  datasourceDb = Object.assign({}, datasourceDb, {
+    url: appEnv.getServiceURL('myCloudant'),
+  });
+  app.bind('datasources.config.db').to(datasourceDb);
+}
+
+export async function main(options?: ApplicationConfig) {
+  // Set the port assigned for the app
+  if (!options) options = {};
+  if (!options.rest) options.rest = {};
+
+  if (appEnv.isLocal) options.rest.port = options.rest.port;
+  else options.rest.port = appEnv.port;
+
+  const app = new TodoListApplication(options);
+  await app.boot();
+  await app.start();
+
+  const url = app.restServer.url;
+  console.log(`Server is running at ${url}`);
+  return app;
+}
+```
+
+## Cloud setup
+
+### Database
+
+1.  Provision a Cloudant database service, and name it `myCloudant`.
+2.  Log on to the Cloudant Dashboard and create a database named `todo`.
+
+### Connecting the database to app
+
+Before we can use the provisioned database, we have to connect the database to
+an app on Cloud Foundry.
+
+So, create a
+[Node.js SDK app](https://console.bluemix.net/catalog/starters/sdk-for-nodejs)
+and connect it to the provisioned Cloudant service.
+
+Make a note of the name of the app. Let's say you named it `mylb4app`. We will
+be using this name as a placeholder for the unique name you will choose yourself
+eventually. Replace `mylb4app` with your app's name in all instances in the doc.
+
+## Deploying the app
+
+### Local
+
+1.  Install all dependencies:
+
+```sh
+npm i
+```
+
+2.  Build the app:
+
+```sh
+npm run build
+```
+
+3.  Ensure Cloudant is running locally.
+4.  Start the app:
+
+```sh
+npm start
+```
+
+5.  Go to http://localhost:3000/swagger-ui to load API Explorer and add a todo
+    or two.
+
+6.  Go to http://localhost:3000/todos to see the todos.
+
+### Cloud
+
+Before we can deploy to IBM Cloud, we need to create two artifacts.
+
+1.  `.cfignore` - not mandatory, prevents `cf` from uploading the listed files
+    to IBM Cloud.
+2.  `manifest.yml` - details about the app.
+
+Create a file named `.cfignore` in the app directory with the following content.
+
+```
+.git/
+node_modules/
+vcap-local.js
+.vscode/
+```
+
+To create the `manifest.yml` file, run `cf create-app-manifest` and specify the
+name of the app that was created earlier `mylb4app`.
+
+```sh
+$ cf create-app-manifest mylb4app -p manifest.yml
+```
+
+Now we are ready to deploy the app to IBM Cloud.
+
+1.  Build the app locally:
+
+```sh
+npm run build
+```
+
+2.  Push the app to Cloud Foundry:
+
+```
+cf push mylb4app
+```
+
```

Comment:
I think there's a missing step of binding the service in the app (from the IBM Cloud console)

Reply:
It is mentioned under "Connecting the database to app".